src/components/movie/MovieDetails.jsx

```diff
@@ -0,0 +1,28 @@
+import { useNavigate, useParams } from "react-router-dom";
+import { GetMoviesBasedOnId, DeleteMovie } from "../../API/movies";
+
+const MovieDetails = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { movies: movie, isPending, error } = GetMoviesBasedOnId(id);
+  const handleDelete = async () => {
+    await DeleteMovie(movie.id).then((resp) => {
+      if (resp.ok) navigate("/");
+    });
+  };
+  return (
+    <div className="movie-details">
+      {movie && (
+        <article>
+          <h2>Title {movie.title}</h2>
```

Comment:
Try to add optional chaining here as well. It's helps us avoid code breakages. Use it especially for values that can be null or undefined. What if there is not a movie.img in the db. Your frontend should have a fallback in place starting with this.

Reply:
Done, I have made these fields required to add in the front end.